clicking manual mixer setup crashes the app

<img width="784" alt="Screenshot 2024-07-31 at 1 48 59 PM" src="https://github.com/user-attachments/assets/726f6ca1-f62b-4149-b5ac-c5c2e4bb9866">


```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x10380d494]

goroutine 45 [running]:
github.com/crypto-power/cryptopower/ui/page/components.(*AccountDropdown).SelectedAccount(...)
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/components/account_dropdown.go:153
github.com/crypto-power/cryptopower/ui/page/privacy.NewManualMixerSetupPage.func2(0x14004b5f800)
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/privacy/manual_mixer_setup_page.go:56 +0x44
github.com/crypto-power/cryptopower/ui/page/components.(*AccountDropdown).Setup(0x1400a2e59a0, {0x1044bc800?, 0x14000970000?})
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/components/account_dropdown.go:52 +0x2f0
github.com/crypto-power/cryptopower/ui/page/privacy.NewManualMixerSetupPage(0x14000e96dc0, 0x14000970000)
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/privacy/manual_mixer_setup_page.go:70 +0x3fc
github.com/crypto-power/cryptopower/ui/page/privacy.(*SetupMixerAccountsPage).HandleUserInteractions(0x140044870e0, {{{0x640, 0x514}, {0x640, 0x514}}, {0x40000000, 0x40000000}, {0xc1a2aa77b1123d98, 0x7880f433f3, 0x1053788a0}, ...})
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/privacy/setup_mixer_accounts_page.go:204 +0x4f0
github.com/crypto-power/cryptopower/ui/page/wallet.(*SingleWalletMasterPage).HandleUserInteractions(0x140052ec6c0, {{{0x640, 0x514}, {0x640, 0x514}}, {0x40000000, 0x40000000}, {0xc1a2aa77b1123d98, 0x7880f433f3, 0x1053788a0}, ...})
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/wallet/single_wallet_main_page.go:287 +0xf8
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).HandleUserInteractions(0x14000503d40, {{{0x640, 0x514}, {0x640, 0x514}}, {0x40000000, 0x40000000}, {0xc1a2aa77b1123d98, 0x7880f433f3, 0x1053788a0}, ...})
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/page/root/home_page.go:370 +0xf8
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0x14000a83b00, {{0xc1a2aa77b1123d98, 0x7880f433f3, 0x1053788a0}, {0x40000000, 0x40000000}, {0x640, 0x514}, {0x0, 0x0, ...}, ...})
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/window.go:252 +0x1e0
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0x14000a83b00)
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/ui/window.go:222 +0x23c
main.main.func4()
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/main.go:121 +0x20
created by main.main in goroutine 1
        /Users/kennedy/go/src/github.com/crypto-power/cryptopower/main.go:119 +0x3a4
exit status 2
```

--- a/ui/page/components/account_dropdown.go
+++ b/ui/page/components/account_dropdown.go
@@ -82,6 +82,11 @@ func (d *AccountDropdown) selectedIsValid() bool {
 	if d.selectedAccount == nil {
 		return false
 	}
+
+	if d.selectedWallet.GetWalletID() != d.selectedAccount.WalletID {
+		return false
+	}
+
 	if d.accountIsValid != nil {
 		if !d.accountIsValid(d.selectedAccount) {
 			return false
@@ -150,6 +155,9 @@ func (d *AccountDropdown) getAccountByNumber(accountNumber int32) *sharedW.Accou
 }
 
 func (d *AccountDropdown) SelectedAccount() *sharedW.Account {
+	if d == nil {
+		return nil
+	}
 	return d.selectedAccount
 }
 

--- a/ui/page/privacy/manual_mixer_setup_page.go
+++ b/ui/page/privacy/manual_mixer_setup_page.go
@@ -137,7 +137,7 @@ func (pg *ManualMixerSetupPage) Layout(gtx C) D {
 				// "Mixed account" label, 4px space, mixed account dropdown
 				layout.Rigid(func(gtx C) D {
 					return layout.Inset{Bottom: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
-						return pg.mixedAccountSelector.Layout(gtx, values.StrMixedAccount)
+						return pg.mixedAccountSelector.Layout(gtx, values.String(values.StrMixedAccount))
 					})
 				}),
 
@@ -151,7 +151,7 @@ func (pg *ManualMixerSetupPage) Layout(gtx C) D {
 				// "Unmixed account" label, 4px space, unmixed account dropdown
 				layout.Rigid(func(gtx C) D {
 					return layout.Inset{Bottom: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
-						return pg.mixedAccountSelector.Layout(gtx, values.StrUnmixedAccount)
+						return pg.unmixedAccountSelector.Layout(gtx, values.String(values.StrUnmixedAccount))
 					})
 				}),
 				// 24px space, then warning/caution text

--- a/ui/page/send/page.go
+++ b/ui/page/send/page.go
@@ -187,6 +187,18 @@ func (pg *Page) pageFields() pageFields {
 
 // initWalletSelector is used for the send modal for wallet selection.
 func (pg *Page) initModalWalletSelector(wallet sharedW.Asset) {
+	pg.walletDropdown = components.NewWalletDropdown(pg.Load).
+		SetChangedCallback(func(wallet sharedW.Asset) {
+			pg.selectedWallet = wallet
+			if pg.accountDropdown != nil {
+				pg.accountDropdown.Setup(wallet)
+				go pg.feeRateSelector.UpdatedFeeRate(pg.selectedWallet)
+				pg.setAssetTypeForRecipients()
+			}
+
+		}).
+		Setup()
+
 	pg.accountDropdown = components.NewAccountDropdown(pg.Load).
 		SetChangedCallback(func(account *sharedW.Account) {
 			pg.initAccountsSelectorForRecipients(account)
@@ -219,18 +231,8 @@ func (pg *Page) initModalWalletSelector(wallet sharedW.Asset) {
 			}
 
 			return accountIsValid
-		})
-	pg.walletDropdown = components.NewWalletDropdown(pg.Load).
-		SetChangedCallback(func(wallet sharedW.Asset) {
-			pg.selectedWallet = wallet
-			if pg.accountDropdown != nil {
-				pg.accountDropdown.Setup(wallet)
-				go pg.feeRateSelector.UpdatedFeeRate(pg.selectedWallet)
-				pg.setAssetTypeForRecipients()
-			}
-
 		}).
-		Setup()
+		Setup(wallet)
 
 	pg.selectedWallet = pg.walletDropdown.SelectedWallet()
 	if wallet != nil {

--- a/ui/page/send/page.go
+++ b/ui/page/send/page.go
@@ -391,10 +391,10 @@ func (pg *Page) constructTx() {
 		pg.feeRateSelector.USDExchangeSet = true
 		pg.txFeeUSD = fmt.Sprintf("$%.4f", utils.CryptoToUSD(pg.exchangeRate, feeAndSize.Fee.CoinValue))
 		pg.feeRateSelector.TxFeeUSD = pg.txFeeUSD
-		pg.totalCostUSD = utils.FormatAsUSDString(pg.Printer, utils.CryptoToUSD(pg.exchangeRate, totalCost.ToCoin() /*totalSendingAmount.ToCoin()*/))
+		pg.totalCostUSD = utils.FormatAsUSDString(pg.Printer, utils.CryptoToUSD(pg.exchangeRate, totalCost.ToCoin()))
 		pg.balanceAfterSendUSD = utils.FormatAsUSDString(pg.Printer, utils.CryptoToUSD(pg.exchangeRate, balanceAfterSend.ToCoin()))
 
-		usdAmount := utils.CryptoToUSD(pg.exchangeRate, wal.ToAmount( /*amountAtom*/ totalAmount).ToCoin())
+		usdAmount := utils.CryptoToUSD(pg.exchangeRate, wal.ToAmount(totalAmount).ToCoin())
 		pg.sendAmountUSD = utils.FormatAsUSDString(pg.Printer, usdAmount)
 	}
 }

--- a/ui/page/send/recipient.go
+++ b/ui/page/send/recipient.go
@@ -104,6 +104,9 @@ func (rp *recipient) initializeAccountSelectors(sourceAccount *sharedW.Account) 
 
 func (rp *recipient) isShowSendToWallet() bool {
 	sourceWalletSelected := rp.sendDestination.walletDropdown.SelectedWallet()
+	if sourceWalletSelected == nil {
+		return false
+	}
 	var wallets []sharedW.Asset
 	switch sourceWalletSelected.GetAssetType() {
 	case libUtil.BTCWalletAsset:

--- a/ui/page/send/recipient.go
+++ b/ui/page/send/recipient.go
@@ -146,11 +146,11 @@ func (rp *recipient) isSendToAddress() bool {
 func (rp *recipient) isValidated() bool {
 	amountIsValid := rp.amount.amountIsValid()
 	addressIsValid := rp.sendDestination.validate()
-
 	// No need for checking the err message since it is as result of amount and
 	// address validation.
 	// validForSending
 	return amountIsValid && addressIsValid
+
 }
 
 func (rp *recipient) resetFields() {

--- a/ui/page/send/send_amount.go
+++ b/ui/page/send/send_amount.go
@@ -219,39 +219,24 @@ func (sa *sendAmount) handle(gtx C) {
 		sa.usdAmountEditor.CustomButton.Background = sa.theme.Color.Gray1
 	}
 
-	for {
-		event, ok := sa.amountEditor.Editor.Update(gtx)
-		if !ok {
-			break
-		}
-
-		if gtx.Source.Focused(sa.amountEditor.Editor) {
-			switch event.(type) {
-			case widget.ChangeEvent:
-				if sa.sendMaxChangeEvent {
-					sa.sendMaxChangeEvent = false
-					continue
-				}
+	if gtx.Source.Focused(sa.amountEditor.Editor) {
+		if sa.amountEditor.Changed() {
+			if sa.sendMaxChangeEvent {
+				sa.sendMaxChangeEvent = false
+			} else {
 				sa.SendMax = false
 				sa.validateAmount()
 				sa.amountChanged()
 			}
+
 		}
 	}
 
-	for {
-		event, ok := sa.usdAmountEditor.Editor.Update(gtx)
-		if !ok {
-			break
-		}
-
-		if gtx.Source.Focused(sa.usdAmountEditor.Editor) {
-			switch event.(type) {
-			case widget.ChangeEvent:
-				if sa.usdSendMaxChangeEvent {
-					sa.usdSendMaxChangeEvent = false
-					continue
-				}
+	if gtx.Source.Focused(sa.usdAmountEditor.Editor) {
+		if sa.usdAmountEditor.Changed() {
+			if sa.usdSendMaxChangeEvent {
+				sa.usdSendMaxChangeEvent = false
+			} else {
 				sa.SendMax = false
 				sa.validateUSDAmount()
 				sa.amountChanged()

--- a/ui/page/send/send_amount.go
+++ b/ui/page/send/send_amount.go
@@ -107,9 +107,8 @@ func (sa *sendAmount) amountIsValid() bool {
 		// do not overwrite existing errors
 		sa.amountErrorText = values.String(values.StrInvalidAmount)
 	}
-
 	amountEditorErrors := sa.amountErrorText == ""
-	return err == nil && amountEditorErrors
+	return err == nil && amountEditorErrors || sa.SendMax
 }
 
 func (sa *sendAmount) validAmount() (int64, bool, error) {

--- a/ui/page/send/send_destination.go
+++ b/ui/page/send/send_destination.go
@@ -58,6 +58,18 @@ func (dst *destination) initDestinationWalletSelector(assetType libUtil.AssetTyp
 				_ = dst.accountDropdown.Setup(wallet)
 			}
 		}).
+		WalletValidator(func(wallet sharedW.Asset) bool {
+			if dst.sourceAccount == nil {
+				return true
+			}
+			if wallet.GetWalletID() == dst.sourceAccount.WalletID {
+				account, err := wallet.GetAccountsRaw()
+				if err != nil || len(account.Accounts) <= 2 {
+					return false
+				}
+			}
+			return true
+		}).
 		EnableWatchOnlyWallets(true).
 		Setup()
 	dst.accountDropdown = components.NewAccountDropdown(dst.Load).

